datafusion/expr-common/src/type_coercion/binary.rs

```diff
@@ -1138,27 +1138,44 @@ fn numeric_string_coercion(lhs_type: &DataType, rhs_type: &DataType) -> Option<D
     }
 }
 
+fn coerce_list_children(lhs_field: &FieldRef, rhs_field: &FieldRef) -> Option<FieldRef> {
+    Some(Arc::new(
+        Arc::unwrap_or_clone(Arc::clone(lhs_field)).with_data_type(comparison_coercion(
```

Comment:
`Arc::unwrap_or_clone` feels unnecessarly indirect, given the fact we know for sure that there are 2+ arcs to the field.
Use ordinary clone